misk-service/src/main/kotlin/misk/ReadyService.kt

```diff
@@ -0,0 +1,31 @@
+package misk
+
+import com.google.common.util.concurrent.AbstractIdleService
+import wisp.logging.getLogger
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * JettyService --> Ready Service --> Feature Flags, Database, AWS, etc
+ *
+ * Thus, we can force JettyService (or SQS, or tasks, or kinesis) to stop only once all the
+ * core infra services have stopped!
+ *
+ * Services that need everything to be ready other than themselves should depend on ReadyService
+ *
+ * Services that are more base should be enhancedBy ReadyService
+ */
+@Singleton
+class ReadyService @Inject constructor() : AbstractIdleService() {
```

Comment:
Different name perhaps?

Reply:
I think the docs need more . . . explanation of this as a symbolic service, useful to depend on or have depend on you?

Reply:
Yeah, you're definitely right.

How's 
```
/**
 * This is a symbolic service that's useful to define the relationship, generally, between
 * services which process traffic (Jetty, SQS, Kinesis, Cron, Tasks, etc.) and services which
 * are required to do work (Database, Redis, GCP, Feature Flags).
 *
 * By having the former depend on ReadyService and the latter enhance ReadyService we can force,
 * for example, JettyService to stop _before_ our feature flag service without having to intertwine
 * our dependency graph.
 */
```

Reply:
That's a nice improvement.